projects/rocprofiler-sdk/source/share/rocprofiler-sdk-rocpd/data_views.sql

```diff
@@ -724,3 +724,85 @@ GROUP BY
     K.dispatch_id,
     PMC_I.name,
     K.agent_id;
+
+CREATE VIEW IF NOT EXISTS
+    `pc_sampling` AS
+SELECT
+    S.id,
+    S.guid,
+    JSON_EXTRACT(S.extdata, '$.sampling_method') AS sampling_method,
+    T.nid,
+    P.pid,
+    TH.tid,
+    S.timestamp,
+    JSON_EXTRACT(S.extdata, '$.instruction') AS instruction,
+    JSON_EXTRACT(S.extdata, '$.instruction_comment') AS instruction_comment,
+    JSON_EXTRACT(E.line_info, '$.pc.code_object_id') AS code_object_id,
+    JSON_EXTRACT(E.line_info, '$.pc.code_object_offset') AS code_object_offset,
+    JSON_EXTRACT(E.extdata, '$.dispatch_id') AS dispatch_id,
+    E.stack_id AS corr_id_internal,
+    E.correlation_id AS corr_id_external,
+    JSON_EXTRACT(S.extdata, '$.exec_mask') AS exec_mask,
+    JSON_EXTRACT(E.extdata, '$.workgroup_id.x') AS workgroup_id_x,
+    JSON_EXTRACT(E.extdata, '$.workgroup_id.y') AS workgroup_id_y,
+    JSON_EXTRACT(E.extdata, '$.workgroup_id.z') AS workgroup_id_z,
+    JSON_EXTRACT(S.extdata, '$.wave_in_group') AS wave_in_group,
+    JSON_EXTRACT(S.extdata, '$.hw_id.chiplet') AS hw_id_chiplet,
+    JSON_EXTRACT(S.extdata, '$.hw_id.wave_id') AS hw_id_wave_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.simd_id') AS hw_id_simd_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.pipe_id') AS hw_id_pipe_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.cu_or_wgp_id') AS hw_id_cu_or_wgp_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.shader_array_id') AS hw_id_shader_array_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.shader_engine_id') AS hw_id_shader_engine_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.workgroup_id') AS hw_id_workgroup_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.vm_id') AS hw_id_vm_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.queue_id') AS hw_id_queue_id,
+    JSON_EXTRACT(S.extdata, '$.hw_id.microengine_id') AS hw_id_microengine_id,
+    -- Stochastic-specific fields (NULL for host_trap)
+    JSON_EXTRACT(E.extdata, '$.wave_issued') AS wave_issued,
+    JSON_EXTRACT(E.extdata, '$.inst_type') AS inst_type,
+    JSON_EXTRACT(E.extdata, '$.wave_count') AS wave_count,
+    JSON_EXTRACT(E.extdata, '$.flags.has_mem_cnt') AS has_mem_cnt,
+    S.event_id,
+    S.track_id,
+    E.stack_id,
+    E.parent_stack_id,
+    E.correlation_id,
+    S.extdata AS sample_extdata,
+    E.extdata AS event_extdata,
+    JSON_EXTRACT(E.extdata, '$.snapshot.stall_reason') AS snapshot_stall_reason,
```

Comment:
Nit-picking, I'm thinking about calling this simply stall_reason.

```suggestion
    JSON_EXTRACT(E.extdata, '$.snapshot.stall_reason') AS stall_reason,
```